runtime/testy/stateManager.py

```diff
@@ -64,24 +66,29 @@ def addPipe(self, processName, pipe):
 
   def createKey(self, keys):
     currDict = self.state
+    path = []
     for key in keys:
       try:
         if key not in currDict:
-          currDict[key] = {}
-        currDict = currDict[key]
+          currDict[key] = [{}, 0]
+        path.append(currDict[key])
+        currDict = currDict[key][0]
       except TypeError:
         error = StudentAPIKeyError(
           "key '{}' is defined, but does not contain a dictionary.".format(key))
         self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(error)
         return
+    currTime = time.time()
+    for item in path:
```

Comment:
@yizow The reason I chose to update timestamps by iterating over a second time was that in the case of an error being thrown while attempting to create a key or set a value, I figured that the timestamps should not be updated.  So I iterate through the first time to create a key or set a value; and then after that loop is done, I know that it was successful, so I then update the timestamps.  If I were setting the timestamps while doing whatever the main operation is, wouldn't I have to write some weird timestamp reversal function in case of an error occurring?